src/NServiceBus.Ninject/NinjectBuilder.cs

```diff
@@ -17,15 +17,25 @@ public class NinjectBuilder : ContainerDefinition
         /// <returns>The new container wrapper.</returns>
         public override ObjectBuilder.Common.IContainer CreateContainer(ReadOnlySettings settings)
         {
-            IKernel existingKernel;
+            KernelHolder kernelHolder;
 
-            if (settings.TryGet("ExistingKernel", out existingKernel))
+            if (settings.TryGet(out kernelHolder))
             {
-                return new NinjectObjectBuilder(existingKernel);
+                return new NinjectObjectBuilder(kernelHolder.ExistingKernel);
 
             }
 
             return new NinjectObjectBuilder();
         }
+
+        internal class KernelHolder
```

Comment:
I think this would not be necessary since the settings holder in v5 doesn't dispose values but it doesn't hurt either.